pulsar-broker/src/test/java/org/apache/pulsar/broker/service/BatchMessageTest.java

```diff
@@ -66,6 +68,7 @@
     @BeforeClass
     @Override
     protected void setup() throws Exception {
+        conf.setAcknowledgmentAtBatchIndexLevelEnabled(true);
```

Comment:
Here here two problems:
- this line affects all the tests in this class, so you are changing existing tests
prwe are not testing what happens when the value is 'false' (so some of your new branches won't be touched by this test)

So probably it is better to run this class with both 'true' and 'false'

Reply:
Ok, I have added a new class with `setAcknowledgmentAtBatchIndexLevelEnabled(true);` to cover these methods.